core/mbed/source/interrupts.cpp

```diff
@@ -26,6 +26,11 @@ void vIRQ_SetVectorX(uint32_t irqn, uint32_t vector, uint32_t flag)
     }
 }
 
+void vIRQ_SetVector(uint32_t irqn, uint32_t vector)
```

Comment:
Yes, since it was for testing purposes having a call gateway to `vIRQ_SetVectorX` is fine as well. The point of the PR was that at the moment `vIRQ_SetVector` is much more widely used but it is not an explicit symbol, which might cause problem in some cases (secure gateway was just an example; debugging is probably another).

In general I'm happier to avoid pre-processors redefinitions, especially for a widely used API.

As for the specific secure gateway example with `vIRQ_SetVector`, it would make sure that a box has exclusive access to that IRQ, for example when running an initialization function:

```C
void box1_init(void)
{
    secure_gateway(box1, vIRQ_SetVector, TRNG_IRQn, (uint32_t) trng_handler);

    /* no need for the rest of the function to run in the context of box 1 */
    ...
}
```
Without the symbol one has to write a wrapper function that calls `vIRQ_SetVector`. Not a big deal, it's just inconsistent with the other APIs, which are available for direct gatewaying.